moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -248,25 +248,27 @@ void PlanningSceneMonitor::initialize(const planning_scene::PlanningScenePtr& sc
   new_scene_update_ = UPDATE_NONE;
 
   last_update_time_ = last_robot_motion_time_ = clock.now();
-  // last_robot_state_update_wall_time_.now() = std::chrono::system_clock::now();
+  last_robot_state_update_wall_time_ = clock.now();
```

Comment:
@davetcoleman and @mlautman 

Im trying to change the ros::wallTime and ros::wallDuration, as was spoken with williams they doesn't seems to be ported to ros2. But the correct way to use it is by replacing ``std::chrono::system_clock::now()`` to ros::wallTime (or at least, this is what I underestood from https://github.com/ros-planning/moveit2/issues/31), also, I'm replacing ros::wallDuration by ``std::chrono::duration`` (such as std:chrono::seconds ...) .

Since there are a lot of time/walltime I would like to know if my procedure with that is the correct one, and if not, please help me with it.

Reply:
This sounds reasonable to me. You should be able to create a quick prototype demonstrating that `ros::WallTime` acts equivalently to `std::chrono::system_clock`. Once you have proved that these are equivalent, or come up with a simple translation, it should be straightforward to do a find-replace across the code base. Note that you also need to update the includes accordingly. I would be more careful with ros::Time as it behaves differently if you are using simulated time vs real time and according to [this ROS Answers post](https://answers.ros.org/question/287946/ros-2-time-handling/) you need a node to be able to get simulated time which may require non-trivial modifications to the code

Reply:
After some testing, I'm going to be using the next code to replace the wallTime and to obtein the sec.

```
    auto time = std::chrono::system_clock::now();
    auto sec = std::chrono::duration<double>(time.time_since_epoch()).count();
``` 
Is it allright?

Reply:
I would discourage the use of auto but otherwise it looks reasonable. I would encourage you to create tests before spending too much time on implementation.